Support Jazzy Jalisco

ROS 2 Jazzy will be released by the end of May. We need to discuss what we want to do, I think the options are:
1. Ignore it for this release (2024.7.0) - release humble only
2. Move main to be based on jazzy and release only jazzy going forward
3. Branch humble to it's own branch, move main to jazzy and release jazzy, continuing to update / release humble when there's a new sync release, but using main (jazzy) for new development

> > How can we check? what would upgrading to jazzy actually involve for `space-ros`? switch to ros2.repos from jazzy?
> 
> We set the ubuntu and ros version variables in the Earthly file and build the main space-ros image. That should "just work"™️ (and if it doesn't, that's a bug) but then it will get interesting when we move into the demos.
> 

Oh, I did not realise it is that easy. Out of interest I just tried building space-ros using ubuntu24 and ROS_DISTRO=jazzy.

it fails on ikos (rosdep stage):

```
             +rosdep | In file included from /opt/spaceros/ikos/core/include/ikos/core/adt/patricia_tree/utils.hpp:48,
             +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/adt/patricia_tree/set.hpp:57,
             +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/linear_constraint.hpp:51,
             +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/value/numeric/interval.hpp:55,
             +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/value/machine_int/interval.hpp:50,
             +rosdep |                  from /opt/spaceros/ikos/ar/include/ikos/ar/pass/simplify_upcast_comparison.hpp:49,
             +rosdep |                  from /opt/spaceros/ikos/ar/src/pass/simplify_upcast_comparison.cpp:44:
             +rosdep | /opt/spaceros/ikos/core/include/ikos/core/semantic/indexable.hpp:56:20: error: ‘uintptr_t’ in namespace ‘std’ does not name a type
             +rosdep |    56 | using Index = std::uintptr_t;
             +rosdep |       |                    ^~~~~~~~~
             +rosdep | /opt/spaceros/ikos/core/include/ikos/core/semantic/indexable.hpp:1:1: note: ‘std::uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
```

I did not look into details, but I presume it is due to compilers versions being different.

> > > How can we check? what would upgrading to jazzy actually involve for `space-ros`? switch to ros2.repos from jazzy?
> > 
> > 
> > We set the ubuntu and ros version variables in the Earthly file and build the main space-ros image. That should "just work"™️ (and if it doesn't, that's a bug) but then it will get interesting when we move into the demos.
> 
> Oh, I did not realise it is that easy. Out of interest I just tried building space-ros using ubuntu24 and ROS_DISTRO=jazzy.
> 
> it fails on ikos (rosdep stage):
> 
> ```
>              +rosdep | In file included from /opt/spaceros/ikos/core/include/ikos/core/adt/patricia_tree/utils.hpp:48,
>              +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/adt/patricia_tree/set.hpp:57,
>              +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/linear_constraint.hpp:51,
>              +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/value/numeric/interval.hpp:55,
>              +rosdep |                  from /opt/spaceros/ikos/core/include/ikos/core/value/machine_int/interval.hpp:50,
>              +rosdep |                  from /opt/spaceros/ikos/ar/include/ikos/ar/pass/simplify_upcast_comparison.hpp:49,
>              +rosdep |                  from /opt/spaceros/ikos/ar/src/pass/simplify_upcast_comparison.cpp:44:
>              +rosdep | /opt/spaceros/ikos/core/include/ikos/core/semantic/indexable.hpp:56:20: error: ‘uintptr_t’ in namespace ‘std’ does not name a type
>              +rosdep |    56 | using Index = std::uintptr_t;
>              +rosdep |       |                    ^~~~~~~~~
>              +rosdep | /opt/spaceros/ikos/core/include/ikos/core/semantic/indexable.hpp:1:1: note: ‘std::uintptr_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
> ```
> 
> I did not look into details, but I presume it is due to compilers versions being different. (there are more compile errors, this is just first one)

may be same issue as https://github.com/NASA-SW-VnV/ikos/issues/273

but not sure why it works in ubuntu22 Earthfile...

Houston, we may have a problem...

I keep trying to build Jazzy/Noble combination after manual fix to ikos.

here are the things I found to fix various errors at different times:

```
sudo apt install libacl1-dev -y
sudo apt install libtinyxml2-dev
sudo apt install libeigen3-dev
sudo apt install python3-lark
sudo apt install python3-numpy
```

I also have to skip *cobra_vendor* because not sure how to fix it:

```
cobra_lib.c: In function ‘load_map’:
cobra_lib.c:1451:52: error: ‘%s’ directive output may be truncated writing up to 1021 bytes into a region of size between 2 and 1023 [-Werror=format-truncation=]
 1451 |                 {       snprintf(a, sizeof(a), "%s/%s", C_BASE, s);
      |                                                    ^~
In file included from /usr/include/stdio.h:980,
                 from cobra_fe.h:12,
                 from cobra.h:10,
                 from cobra_lib.c:7:
In function ‘snprintf’,
    inlined from ‘load_map’ at cobra_lib.c:1451:5:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: ‘__builtin___snprintf_chk’ output between 2 and 2044 bytes into a destination of size 1024
   54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   55 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   56 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [<builtin>: cobra_lib.o] Error 1
gmake[2]: *** [CMakeFiles/cobra-4.1.dir/build.make:86: cobra-4.1/src/cobra-4.1-stamp/cobra-4.1-build] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/cobra-4.1.dir/all] Error 2
gmake: *** [Makefile:134: all] Error 2
```

I also have to skip *python_orocos_kdl_vendor* because of:

```
In file included from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/cast.h:16,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/attr.h:13,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/pybind11.h:13,
                 from /opt/spaceros/build/python_orocos_kdl_vendor/_deps/python_orocos_kdl-src/python_orocos_kdl/PyKDL/PyKDL.h:26,
                 from /opt/spaceros/build/python_orocos_kdl_vendor/_deps/python_orocos_kdl-src/python_orocos_kdl/PyKDL/PyKDL.cpp:27:
/opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/type_caster_base.h: In function ‘std::string pybind11::detail::error_string()’:
/opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/type_caster_base.h:482:26: error: invalid use of incomplete type ‘PyFrameObject’ {aka ‘struct _frame’}
  482 |             frame = frame->f_back;
      |                          ^~
In file included from /usr/include/python3.12/Python.h:42,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/common.h:215,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/pytypes.h:12,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/cast.h:13:
/usr/include/python3.12/pytypedefs.h:22:16: note: forward declaration of ‘PyFrameObject’ {aka ‘struct _frame’}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
Summary: 108 packages finished [54.8s]
  1 package failed: python_orocos_kdl_vendor
```

but then this bind11 issue happens in *rclpy*, at which point I am not sure what to do:

```
...
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:244: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/graph.cpp.o] Error 1
In file included from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/cast.h:16,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/attr.h:13,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/pybind11.h:13,
                 from /opt/spaceros/src/rclpy/rclpy/src/rclpy/lifecycle.cpp:15:
/opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/type_caster_base.h: In function ‘std::string pybind11::detail::error_string()’:
/opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/type_caster_base.h:482:26: error: invalid use of incomplete type ‘PyFrameObject’ {aka ‘struct _frame’}
  482 |             frame = frame->f_back;
      |                          ^~
In file included from /usr/include/python3.12/Python.h:42,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/detail/common.h:215,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/pytypes.h:12,
                 from /opt/spaceros/install/pybind11_vendor/include/pybind11_vendor/pybind11/cast.h:13:
/usr/include/python3.12/pytypedefs.h:22:16: note: forward declaration of ‘PyFrameObject’ {aka ‘struct _frame’}
   22 | typedef struct _frame PyFrameObject;
      |                ^~~~~~
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:258: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/guard_condition.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:132: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/action_server.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:202: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/duration.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:188: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/destroyable.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:90: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/_rclpy_pybind11.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:216: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/clock_event.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:286: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/logging.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:174: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/context.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/_rclpy_pybind11.dir/build.make:272: CMakeFiles/_rclpy_pybind11.dir/src/rclpy/lifecycle.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:212: CMakeFiles/_rclpy_pybind11.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< rclpy [37.3s, exited with code 2]
Aborted  <<< rclcpp [2min 10s]                                            
```

What I do not understand is how it works with standard ROS2 Jazzy build from source, as recommended (https://docs.ros.org/en/jazzy/Installation/Alternatives/Ubuntu-Development-Setup.html). I believe we are doing everything same way...

Anyone wants to reproduce or has ideas how to move forward?

--- a/Earthfile
+++ b/Earthfile
@@ -11,20 +11,19 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-VERSION 0.6
-FROM ubuntu:jammy
+VERSION 0.8
+FROM ubuntu:noble
 
-# Defaulting to ROS 2 humble
-ARG ROS_DISTRO="humble"
+# Defaulting to ROS 2 jazzy
+ARG ROS_DISTRO="jazzy"
+
+# TODO - the `setup` step will be merged with the `setup` step in spaceros docker Earthfile
+# This variable will then act as a single source of truth.
+ENV ROS_DISTRO ${ROS_DISTRO}
 
 setup:
   # Disable prompting during package installation
   ARG DEBIAN_FRONTEND=noninteractive
-  ARG ROS_DISTRO
-
-  # TODO - the `setup` step will be merged with the `setup` step in spaceros docker Earthfile
-  # This variable will then act as a single source of truth.
-  ENV ROS_DISTRO ${ROS_DISTRO}
 
   # The following commands are based on the source install for ROS 2 Rolling Ridley.
   # See: https://docs.ros.org/en/ros2_documentation/rolling/Installation/Ubuntu-Development-Setup.html
@@ -171,13 +170,16 @@ rosdep:
       --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
       sudo apt-get update && \
       rosdep update && \
-      rosdep install -y \
+      rosdep install -s -y \
         --from-paths src --ignore-src \
         --rosdistro ${ROS_DISTRO} \
         # `urdfdom_headers` is cloned from source, however rosdep can't find it.
         # It is because package.xml manifest is missing. See: https://github.com/ros/urdfdom_headers
         # Additionally, IKOS must be excluded as per: https://github.com/space-ros/docker/issues/99
-        --skip-keys "$(tr '\n' ' ' < 'excluded-pkgs.txt') urdfdom_headers ikos"
+        --skip-keys "$(tr '\n' ' ' < 'excluded-pkgs.txt') urdfdom_headers ikos" > rosdep-commands.sh && \
+      chmod u+x rosdep-commands.sh && \
+      ./rosdep-commands.sh
+        
   RUN rm excluded-pkgs.txt
 
   RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
@@ -257,7 +259,7 @@ build-testing:
 image:
   FROM +rosdep
   ARG VCS_REF
-  ARG VERSION="latest"
+  ARG VERSION='jazzy'
 
   # Specify the docker image metadata
   LABEL org.label-schema.schema-version="1.0"

--- a/Earthfile
+++ b/Earthfile
@@ -234,6 +234,8 @@ build:
 
 build-dev:
   FROM +rosdep
+  ARG tag='jazzy'
+
   RUN colcon build \
       --cmake-args \
       -DCMAKE_BUILD_TYPE=RelWithDebInfo \
@@ -245,9 +247,11 @@ build-dev:
 
 build-testing:
   FROM +build-dev
+  # exclude ament_lint tests as they give error on Jazzy due to python 'unittest' requiring at least one test recently
   RUN . install/setup.sh && \
       colcon test \
         --retest-until-pass 2 \
+        --packages-skip ament_lint \
         --ctest-args -LE "(ikos|xfail)" \
         --pytest-args -m "not xfail"
   RUN . install/setup.sh && \

--- a/Earthfile
+++ b/Earthfile
@@ -207,7 +207,7 @@ rosdep:
         clang-14
 
   WORKDIR ${SPACEROS_DIR}
-  RUN git clone -b v3.2 --depth 1 https://github.com/NASA-SW-VnV/ikos.git
+  RUN git clone --branch v3.4 --depth 1 https://github.com/NASA-SW-VnV/ikos.git
   WORKDIR ${SPACEROS_DIR}/ikos
   RUN mkdir build
   WORKDIR ${SPACEROS_DIR}/ikos/build
@@ -216,7 +216,7 @@ rosdep:
         -DCMAKE_BUILD_TYPE="Debug" \
         -DLLVM_CONFIG_EXECUTABLE="/usr/lib/llvm-14/bin/llvm-config" \
         ..
-  RUN make
+  RUN make -j`nproc`
   RUN sudo make install
   ENV PATH="/opt/ikos/bin/:$PATH"
   WORKDIR ${SPACEROS_DIR}

--- a/docker/scripts/generate-repos.sh
+++ b/docker/scripts/generate-repos.sh
@@ -11,7 +11,7 @@ usage() {
     echo "  --repos                     File containing a list of ROS repos to use."
     echo "                              Use as alternative to 'packages', rosinstall can generate .repos file from list of repos instead of list of packages"
     echo "  --excluded-packages         File containing a list of packages to exclude [optional] "
-    echo "  --rosdistro                 ROS2 distribution (default: humble)"
+    echo "  --rosdistro                 ROS2 distribution (default: jazzy)"
     echo "  --upstream                  Specify whether to use version tags of upstream repositories (default: true)"
     echo "  --exclude-installed         Whether to exclude already installed packages from .repos file. Installed workspaces must be sourced to make it work (default: true)"
     echo "  -h, --help                  Display this help and exit"
@@ -19,7 +19,7 @@ usage() {
 }
 
 # Initialize variables with default values
-rosdistro="humble"
+rosdistro="jazzy"
 packages=""
 outfile=""
 excluded_packages=""

--- a/ikos.repos
+++ b/ikos.repos
@@ -1,5 +1,0 @@
-repositories:
-  ikos:
-    type: git
-    url: https://github.com/NASA-SW-VnV/ikos.git
-    version: v3.2

--- a/ros2.repos
+++ b/ros2.repos
@@ -11,7 +11,7 @@ repositories:
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
-    version: master
+    version: main
   ament_ikos:
     type: git
     url: https://github.com/ament/ament_ikos.git

--- a/ros2.repos
+++ b/ros2.repos
@@ -11,7 +11,7 @@ repositories:
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
-    version: main
+    version: v0.12.0
   ament_ikos:
     type: git
     url: https://github.com/ament/ament_ikos.git

--- a/ros2.repos
+++ b/ros2.repos
@@ -3,11 +3,11 @@ repositories:
   ament_cmake:
     type: git
     url: https://github.com/ament/ament_cmake.git
-    version: 1.3.10
+    version: 2.5.2
   ament_cmake_ros:
     type: git
     url: https://github.com/ros2/ament_cmake_ros.git
-    version: 0.10.0
+    version: 0.12.0
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
@@ -19,7 +19,7 @@ repositories:
   ament_index:
     type: git
     url: https://github.com/ament/ament_index.git
-    version: 1.4.0
+    version: 1.8.1
   ament_lint:
     type: git
     url: https://github.com/ament/ament_lint.git
@@ -27,39 +27,39 @@ repositories:
   ament_package:
     type: git
     url: https://github.com/ament/ament_package.git
-    version: 0.14.0
+    version: 0.16.3
   class_loader:
     type: git
     url: https://github.com/ros/class_loader.git
-    version: 2.2.0
+    version: 2.7.0
   common_interfaces:
     type: git
     url: https://github.com/ros2/common_interfaces.git
-    version: 4.2.4
+    version: 5.3.5
   console_bridge_vendor:
     type: git
     url: https://github.com/ros2/console_bridge_vendor.git
-    version: 1.4.1
+    version: 1.7.1
   cyclonedds:
     type: git
     url: https://github.com/eclipse-cyclonedds/cyclonedds.git
-    version: 0.10.4
+    version: 0.10.5
   eigen3_cmake_module:
     type: git
     url: https://github.com/ros2/eigen3_cmake_module.git
-    version: 0.1.1
+    version: 0.3.0
   geometry2:
     type: git
     url: https://github.com/ros2/geometry2.git
-    version: 0.25.7
+    version: 0.36.4
   google_benchmark_vendor:
     type: git
     url: https://github.com/ament/google_benchmark_vendor.git
-    version: 0.1.2
+    version: 0.5.0
   googletest:
     type: git
     url: https://github.com/ament/googletest.git
-    version: 1.10.9006
+    version: 1.14.9000
   iceoryx:
     type: git
     url: https://github.com/eclipse-iceoryx/iceoryx.git
@@ -67,51 +67,51 @@ repositories:
   kdl_parser:
     type: git
     url: https://github.com/ros/kdl_parser.git
-    version: 2.6.4
+    version: 2.11.0
   launch:
     type: git
     url: https://github.com/ros2/launch.git
-    version: 1.0.6
+    version: 3.4.2
   launch_ros:
     type: git
     url: https://github.com/ros2/launch_ros.git
-    version: 0.19.7
+    version: 0.26.5
   libstatistics_collector:
     type: git
     url: https://github.com/ros-tooling/libstatistics_collector.git
-    version: 1.3.2
+    version: 1.7.3
   libyaml_vendor:
     type: git
     url: https://github.com/ros2/libyaml_vendor.git
-    version: 1.2.2
+    version: 1.6.3
   message_filters:
     type: git
     url: https://github.com/ros2/message_filters.git
-    version: 4.3.4
+    version: 4.11.2
   mimick_vendor:
     type: git
     url: https://github.com/ros2/mimick_vendor.git
-    version: 0.2.8
+    version: 0.6.1
   orocos_kdl_vendor:
     type: git
     url: https://github.com/ros2/orocos_kdl_vendor.git
-    version: 0.2.5
+    version: 0.5.1
   osrf_pycommon:
     type: git
     url: https://github.com/osrf/osrf_pycommon.git
-    version: 2.0.2
+    version: 2.1.4
   osrf_testing_tools_cpp:
     type: git
     url: https://github.com/osrf/osrf_testing_tools_cpp.git
-    version: 1.5.2
+    version: 2.0.0
   performance_test_fixture:
     type: git
     url: https://github.com/ros2/performance_test_fixture.git
-    version: 0.0.9
+    version: 0.2.0
   pluginlib:
     type: git
     url: https://github.com/ros/pluginlib.git
-    version: 5.1.0
+    version: 5.4.2
   process_sarif:
     type: git
     url: https://github.com/space-ros/process_sarif.git
@@ -119,132 +119,136 @@ repositories:
   pybind11_vendor:
     type: git
     url: https://github.com/ros2/pybind11_vendor.git
-    version: 2.4.2
+    version: 3.1.2
   python_cmake_module:
     type: git
     url: https://github.com/ros2/python_cmake_module.git
-    version: 0.10.0
+    version: 0.11.1
   rcl:
     type: git
     url: https://github.com/ros2/rcl.git
-    version: 5.3.9
+    version: 9.2.3
   rcl_interfaces:
     type: git
     url: https://github.com/ros2/rcl_interfaces.git
-    version: 1.2.1
+    version: 2.0.2
   rcl_logging:
     type: git
     url: https://github.com/ros2/rcl_logging.git
-    version: 2.3.1
+    version: 3.1.0
   rclcpp:
     type: git
     url: https://github.com/ros2/rclcpp.git
-    version: 16.0.10
+    version: 28.1.3
   rclpy:
     type: git
     url: https://github.com/ros2/rclpy.git
-    version: 3.3.14
+    version: 7.1.1
   rcpputils:
     type: git
     url: https://github.com/ros2/rcpputils.git
-    version: 2.4.3
+    version: 2.11.0
   rcutils:
     type: git
     url: https://github.com/ros2/rcutils.git
-    version: 5.1.6
+    version: 6.7.1
   rmw:
     type: git
     url: https://github.com/ros2/rmw.git
-    version: 6.1.2
+    version: 7.3.1
   rmw_cyclonedds:
     type: git
     url: https://github.com/ros2/rmw_cyclonedds.git
-    version: 1.3.4
+    version: 2.2.2
   rmw_dds_common:
     type: git
     url: https://github.com/ros2/rmw_dds_common.git
-    version: 1.6.0
+    version: 3.1.0
   rmw_implementation:
     type: git
     url: https://github.com/ros2/rmw_implementation.git
-    version: 2.8.4
+    version: 2.15.3
   robot_state_publisher:
     type: git
     url: https://github.com/ros/robot_state_publisher.git
-    version: 3.0.3
+    version: 3.3.3
   ros2_tracing:
     type: git
     url: https://github.com/ros2/ros2_tracing.git
-    version: 4.1.1
+    version: 8.2.1
   ros2cli:
     type: git
     url: https://github.com/ros2/ros2cli.git
-    version: 0.18.11
+    version: 0.32.1
   ros_environment:
     type: git
     url: https://github.com/ros/ros_environment.git
-    version: 3.2.2
+    version: 4.2.1
   ros_testing:
     type: git
     url: https://github.com/ros2/ros_testing.git
-    version: 0.4.0
+    version: 0.6.0
   rosidl:
     type: git
     url: https://github.com/ros2/rosidl.git
-    version: 3.1.5
+    version: 4.6.3
+  rosidl_core:
+    type: git
+    url: https://github.com/ros2/rosidl_core.git
+    version: 0.2.0
   rosidl_defaults:
     type: git
     url: https://github.com/ros2/rosidl_defaults.git
-    version: 1.2.0
+    version: 1.6.0
+  rosidl_dynamic_typesupport:
+    type: git
+    url: https://github.com/ros2/rosidl_dynamic_typesupport.git
+    version: 0.1.2
   rosidl_python:
     type: git
     url: https://github.com/ros2/rosidl_python.git
-    version: 0.14.4
+    version: 0.22.0
   rosidl_runtime_py:
     type: git
     url: https://github.com/ros2/rosidl_runtime_py.git
-    version: 0.9.3
+    version: 0.13.1
   rosidl_typesupport:
     type: git
     url: https://github.com/ros2/rosidl_typesupport.git
-    version: 2.0.1
+    version: 3.2.2
   rpyutils:
     type: git
     url: https://github.com/ros2/rpyutils.git
-    version: 0.2.1
+    version: 0.4.1
   spdlog_vendor:
     type: git
     url: https://github.com/ros2/spdlog_vendor.git
-    version: 1.3.1
+    version: 1.6.1
   test_interface_files:
     type: git
     url: https://github.com/ros2/test_interface_files.git
-    version: 0.9.1
+    version: 0.11.0
   tinyxml2_vendor:
     type: git
     url: https://github.com/ros2/tinyxml2_vendor.git
-    version: 0.7.6
-  tinyxml_vendor:
-    type: git
-    url: https://github.com/ros2/tinyxml_vendor.git
-    version: 0.8.3
+    version: 0.9.1
   uncrustify_vendor:
     type: git
     url: https://github.com/ament/uncrustify_vendor.git
-    version: 2.0.2
+    version: 3.0.0
   unique_identifier_msgs:
     type: git
     url: https://github.com/ros2/unique_identifier_msgs.git
-    version: 2.2.1
+    version: 2.5.0
   urdf:
     type: git
     url: https://github.com/ros2/urdf.git
-    version: 2.6.0
+    version: 2.10.0
   urdfdom:
     type: git
     url: https://github.com/ros/urdfdom.git
-    version: 3.0.2
+    version: 4.0.0
   urdfdom_headers:
     type: git
     url: https://github.com/ros/urdfdom_headers.git
-    version: 1.0.6
+    version: 1.1.1

--- a/spaceros.repos
+++ b/spaceros.repos
@@ -4,7 +4,7 @@ repositories:
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
-    version: main
+    version: v0.12.0
   ament_ikos:
     type: git
     url: https://github.com/ament/ament_ikos.git

--- a/spaceros.repos
+++ b/spaceros.repos
@@ -4,7 +4,7 @@ repositories:
   ament_cobra:
     type: git
     url: https://github.com/ament/ament_cobra.git
-    version: master
+    version: main
   ament_ikos:
     type: git
     url: https://github.com/ament/ament_ikos.git

